File: task1/C/task1.cpp
#include <iostream>
#include <math.h>
using namespace :: std;

bool isPrime(int num) {
    if (num <= 1) return false; // Числа меньше или равные 1 не являются простыми.
    for (int i = 2; i <= sqrt(num); ++i) {
        if (num % i == 0) return false;
    }
    return true;
}

int closestPrimeNumber(int num) {
    if (isPrime(num)) return 0;
    for (int i = 1; ; ++i) {
        if (isPrime(num - i) || isPrime(num + i)) return i;
    }
}

int main() {
    int N;
    cout << "Enter an integer: ";
    cin >> N;
    cout << closestPrimeNumber(N) << endl;
    return 0;
}

File: task1/JS/task1.js
function isPrime(num) {
    if (num <= 1) return false;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}

function closestPrimeNumber(num) {
    if (isPrime(num)) return 0;
    let i = 1;
    while (true) {
        if (isPrime(num - i) || isPrime(num + i)) return i;
        i++;
    }
}

// Ввод (Node.js)
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question("Enter an integer: ", (input) => {
    const N = parseInt(input);
    console.log(closestPrimeNumber(N));
    readline.close();
});
